src/View/TemplateEngine.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace SilverStripe\View;
+
+use SilverStripe\View\Exception\MissingTemplateException;
+
+/**
+ * Interface for template rendering engines such as twig or ss templates.
+ */
+interface TemplateEngine
```

Comment:
As mentioned on a comment on SSTemplateEngine, this should probably an be abstract class and lots of the code that's currently on SSTemplateEngine should live here since presumably any other template engine could also use it

Reply:
Discussion about that can be held in https://github.com/silverstripe/silverstripe-framework/pull/11405#discussion_r1800395370 where you raised it initially.

Reply:
All good, leave as is